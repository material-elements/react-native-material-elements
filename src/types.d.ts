export * from './components/types';
export * from './types/index';
