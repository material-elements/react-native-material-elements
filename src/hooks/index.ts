export * from './useRestyle';
export * from './useThemedProps';
