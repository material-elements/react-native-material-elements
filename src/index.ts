export * from './components';
export * from './hooks';
export * from './libraries';
export * from './utils';
