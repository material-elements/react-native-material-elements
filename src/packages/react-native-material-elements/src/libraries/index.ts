export * from './themes';
export * from './style';
