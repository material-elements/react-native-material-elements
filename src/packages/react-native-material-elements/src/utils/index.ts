export * from './styleGenerator';
export * from './utils';
export * from './dimensions';
