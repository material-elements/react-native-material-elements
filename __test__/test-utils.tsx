import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';

export const ThemeWrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: ThemeWrapper, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
