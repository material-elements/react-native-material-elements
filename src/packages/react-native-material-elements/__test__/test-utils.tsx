import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';
import { ThemeProviderProps } from '../src/libraries/types';

interface CustomRenderOptions extends Omit<RenderOptions, 'wrapper'> {
  themeProps?: Partial<ThemeProviderProps>;
}

export const ThemeWrapper: React.FC<ThemeProviderProps> = ({ children, ...props }) => {
  return <ThemeProvider {...props}>{children}</ThemeProvider>;
};

export const customRender = (ui: React.ReactElement, options?: CustomRenderOptions) => {
  const { themeProps, ...rest } = options || {};
  return render(ui, {
    wrapper: wrapperProps => <ThemeWrapper {...themeProps}>{wrapperProps.children}</ThemeWrapper>,
    ...rest,
  });
};

export * from '@testing-library/react-native';
export { customRender as render };
