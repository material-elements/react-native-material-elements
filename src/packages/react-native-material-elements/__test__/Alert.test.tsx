import { StyleSheet, TouchableOpacity, useColorScheme, View } from 'react-native';
import { Alert, gray, Text, yellow } from '../src';
import { fireEvent, render } from './test-utils';
import * as RN from 'react-native';

describe('Alert', () => {
  const mockStartIcon = <View testID="start-icon-test-id" />;
  const mockEndIcon = <View testID="end-icon-test-id" />;

  const mockOnPress = jest.fn();
  const testId = 'alert-container-test-id';

  beforeEach(() => {
    jest.clearAllMocks();
  });

  const mockAction = (
    <TouchableOpacity testID="action-test-id" onPress={mockOnPress}>
      <Text>Click here</Text>
    </TouchableOpacity>
  );

  it('should render correctly with zero props', () => {
    const { toJSON } = render(<Alert />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render start icon correctly', () => {
    const { getByTestId } = render(<Alert startIcon={mockStartIcon} />);
    const startIconElement = getByTestId('start-icon-test-id');
    expect(startIconElement).toBeDefined();
  });

  it('should render end icon correctly', () => {
    const { getByTestId } = render(<Alert endIcon={mockEndIcon} />);
    const endIconElement = getByTestId('end-icon-test-id');
    expect(endIconElement).toBeDefined();
  });

  it('should render the action component correctly', () => {
    const { getByTestId } = render(<Alert action={mockAction} />);
    const actionButton = getByTestId('action-test-id');
    fireEvent.press(actionButton);
    expect(mockOnPress).toHaveBeenCalled();
  });

  it('should render the title', () => {
    const { getByText } = render(<Alert title="Hello" />);
    const titleElement = getByText('Hello');
    expect(titleElement).toBeDefined();
  });

  it('should render the subTitle', () => {
    const { getByText } = render(<Alert subTitle="Hello" />);
    const subTitleElement = getByText('Hello');
    expect(subTitleElement).toBeDefined();
  });

  it('should render the ghost variant alert with (light theme)', () => {
    (useColorScheme as jest.Mock).mockReturnValue('light');
    const { getByTestId } = render(<Alert variation="ghost" testID={testId} />);

    const alert = getByTestId(testId);

    const flattenStyles = StyleSheet.flatten(alert.props.style);
    expect(flattenStyles).toEqual(
      expect.objectContaining({ borderRadius: 6, borderColor: gray[300], borderWidth: 0.8, backgroundColor: gray[100] }),
    );
  });

  it('should render the ghost variant alert with (dark theme)', () => {
    (useColorScheme as jest.Mock).mockReturnValue('dark');
    const { getByTestId } = render(<Alert variation="ghost" testID={testId} />);

    const alert = getByTestId(testId);

    const flattenStyles = StyleSheet.flatten(alert.props.style);
    expect(flattenStyles).toEqual(
      expect.objectContaining({ borderRadius: 6, borderColor: gray[700], borderWidth: 0.8, backgroundColor: gray[800] }),
    );
  });

  it('should render gray title when variation is ghost, variant is gray and theme mode is light', () => {
    (useColorScheme as jest.Mock).mockReturnValue('light');
    const { getByText } = render(<Alert variation="ghost" variant="gray" subTitle="subTitle" title="title" />);

    const title = getByText('title');
    const subTitle = getByText('subTitle');

    const titleFlattenStyles = StyleSheet.flatten(title.props.style);
    expect(titleFlattenStyles).toEqual(expect.objectContaining({ color: gray[800] }));

    const subTitleFlattenStyles = StyleSheet.flatten(subTitle.props.style);
    expect(subTitleFlattenStyles).toEqual(expect.objectContaining({ color: gray[800] }));
  });

  it('should render light gray title when variation is ghost, variant is gray and theme mode is dark', () => {
    (useColorScheme as jest.Mock).mockReturnValue('dark');
    const { getByText } = render(<Alert variation="ghost" variant="gray" subTitle="subTitle" title="title" />);

    const title = getByText('title');
    const subTitle = getByText('subTitle');

    const titleFlattenStyles = StyleSheet.flatten(title.props.style);
    expect(titleFlattenStyles).toEqual(expect.objectContaining({ color: 'white' }));

    const subTitleFlattenStyles = StyleSheet.flatten(subTitle.props.style);
    expect(subTitleFlattenStyles).toEqual(expect.objectContaining({ color: 'white' }));
  });

  it('should render gray title when variation is ghost, variant is lightGray and theme mode is light', () => {
    (useColorScheme as jest.Mock).mockReturnValue('light');
    const { getByText } = render(<Alert variation="ghost" variant="lightGray" subTitle="subTitle" title="title" />);

    const title = getByText('title');
    const subTitle = getByText('subTitle');

    const titleFlattenStyles = StyleSheet.flatten(title.props.style);
    expect(titleFlattenStyles).toEqual(expect.objectContaining({ color: gray[800] }));

    const subTitleFlattenStyles = StyleSheet.flatten(subTitle.props.style);
    expect(subTitleFlattenStyles).toEqual(expect.objectContaining({ color: gray[800] }));
  });

  it('should render light gray title when variation is ghost, variant is lightGray and theme mode is dark', () => {
    (useColorScheme as jest.Mock).mockReturnValue('dark');
    const { getByText } = render(<Alert variation="ghost" variant="lightGray" subTitle="subTitle" title="title" />);

    const title = getByText('title');
    const subTitle = getByText('subTitle');

    const titleFlattenStyles = StyleSheet.flatten(title.props.style);
    expect(titleFlattenStyles).toEqual(expect.objectContaining({ color: 'white' }));

    const subTitleFlattenStyles = StyleSheet.flatten(subTitle.props.style);
    expect(subTitleFlattenStyles).toEqual(expect.objectContaining({ color: 'white' }));
  });

  it('should render gray title when variation is ghost, variant is warning and theme mode is light', () => {
    (useColorScheme as jest.Mock).mockReturnValue('light');
    const { getByText } = render(<Alert variation="ghost" variant="warning" subTitle="subTitle" title="title" />);

    const title = getByText('title');
    const subTitle = getByText('subTitle');

    const titleFlattenStyles = StyleSheet.flatten(title.props.style);
    expect(titleFlattenStyles).toEqual(expect.objectContaining({ color: gray[800] }));

    const subTitleFlattenStyles = StyleSheet.flatten(subTitle.props.style);
    expect(subTitleFlattenStyles).toEqual(expect.objectContaining({ color: gray[800] }));
  });

  it('should render yellow title when variation is ghost, variant is warning and theme mode is dark', () => {
    (useColorScheme as jest.Mock).mockReturnValue('dark');
    const { getByText, debug } = render(<Alert variation="ghost" variant="warning" subTitle="subTitle" title="title" />);

    const title = getByText('title');
    const subTitle = getByText('subTitle');

    const titleFlattenStyles = StyleSheet.flatten(title.props.style);
    expect(titleFlattenStyles).toEqual(expect.objectContaining({ color: yellow[400] }));

    const subTitleFlattenStyles = StyleSheet.flatten(subTitle.props.style);
    expect(subTitleFlattenStyles).toEqual(expect.objectContaining({ color: yellow[400] }));
  });
});
