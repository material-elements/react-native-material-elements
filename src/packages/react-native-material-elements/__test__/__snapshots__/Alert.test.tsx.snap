// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render correctly with zero props 1`] = `
<View
  style={
    [
      {},
      {},
      [
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "minHeight": 30,
          "paddingBottom": 4,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 4,
        },
        {
          "backgroundColor": "#4CAF50",
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  />
</View>
`;
